### PR TITLE
Revise sign

### PR DESCRIPTION
**Pull request by pszmel**
### Details:
- update docs
- add RTTI macro
-  create visitor test
- update backend test 
- create type_prop tests
- add op to constants.py
- add to SLT int tests

### Tickets:
 - 37477

### CODE REVIEW

**Comment by ilyachur** [comment]
@rkazants Please have a look

File: docs/template_plugin/tests/functional/op_reference/sign.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#include "base_reference_test.hpp"
+
+using namespace ngraph;
+using namespace InferenceEngine;
+
+struct SignParams {
+    template <class IT, class OT>
+    SignParams(const ngraph::PartialShape& shape, const ngraph::element::Type& iType, const ngraph::element::Type& oType, const std::vector<IT>& iValues,
+                  const std::vector<OT>& oValues, size_t iSize = 0, size_t oSize = 0)
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/sign.cpp:21]: I don't think that you need to have `iSize`, `oSize`. You always work with default types

File: docs/template_plugin/tests/functional/op_reference/sign.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+TEST_P(ReferenceSignLayerTest, CompareWithHardcodedRefs) {
+    Exec();
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Sign_With_Hardcoded_Refs, ReferenceSignLayerTest,
+    ::testing::Values(
+        // destination boolean
+        SignParams(ngraph::PartialShape {6}, ngraph::element::f32, ngraph::element::f32,
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/sign.cpp:63]: Could you cover all supported data types?

File: docs/template_plugin/tests/functional/op_reference/sign.cpp (hunk 1 above)
Comment (pszmel) [reply docs/template_plugin/tests/functional/op_reference/sign.cpp]: Right, removed.

File: docs/template_plugin/tests/functional/op_reference/sign.cpp (hunk 2 above)
Comment (pszmel) [reply docs/template_plugin/tests/functional/op_reference/sign.cpp:62]: Sure. I added test for all supported data types

Comment (ilyachur) [reply docs/template_plugin/tests/functional/op_reference/sign.cpp:62]: I don't see boolean  type

**Review by ilyachur — COMMENTED** [review]
Should we update serialization tests?

**Review by ilyachur — APPROVED** [review]
@rkazants Could you check specification style?

**Review by rkazants — APPROVED** [review]

**Pull request merged**